clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/P2PBlobTransferManagerFactory.java

```diff
@@ -0,0 +1,136 @@
+package com.linkedin.davinci.blobtransfer;
+
+import static com.linkedin.davinci.blobtransfer.BlobTransferGlobalTrafficShapingHandlerHolder.getGlobalChannelTrafficShapingHandlerInstance;
+
+import com.linkedin.davinci.blobtransfer.client.NettyFileTransferClient;
+import com.linkedin.davinci.blobtransfer.server.P2PBlobTransferService;
+import com.linkedin.davinci.stats.AggVersionedBlobTransferStats;
+import com.linkedin.davinci.storage.StorageEngineRepository;
+import com.linkedin.davinci.storage.StorageMetadataService;
+import com.linkedin.venice.blobtransfer.DaVinciBlobFinder;
+import com.linkedin.venice.blobtransfer.ServerBlobFinder;
+import com.linkedin.venice.client.store.ClientConfig;
+import com.linkedin.venice.helix.HelixCustomizedViewOfflinePushRepository;
+import com.linkedin.venice.meta.ReadOnlyStoreRepository;
+import io.netty.handler.traffic.GlobalChannelTrafficShapingHandler;
+import java.util.concurrent.CompletableFuture;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class P2PBlobTransferManagerFactory {
```

Comment:
Can we convert it into a builder pattern?
Some examples:
https://github.com/linkedin/venice/blob/main/clients/venice-client/src/main/java/com/linkedin/venice/fastclient/ClientConfig.java

With the builder pattern, I think we can merge the following two functions.